Johannes/levelmasks

holke opened the pull request:
AHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHH

Margrit, i need your help.

I just wanted to merge the new master into this branch in order to be able to build on top of the new levels and properly integrate the mask level.

But i somehow totally messed up :(
The player does not enter the house anymore in the get home level.
It does if you press space, but then the small player does not look out of the window and the level does not end :/
The same happens in the supermarket level.
In the mask level also the mini player does not show up anymore when you press space.

Additionally in the supermarket some NPCs have just stopped moving!

HHEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEELP

Edit: please don’t merge. I forgot to set this to draft.

holke committed:
Insert a new level. This  is the base for the mask collect level

holke committed:
Added mask as prefab and mask level start.

The mask rotates.
The mask is created on startup of the level at a random position.

holke committed:
Experiment with mask colliders. WIP

holke committed:
Merge remote-tracking branch 'origin/master' into johannes/levelmasks

holke committed:
The mask now finally collides with the player and then spawns at new position

holke committed:
The player now keeps an inventory of masks

holke committed:
Add subfolders for level. Add text that counts masks

holke committed:
Add display for number of collected masks

holke committed:
First working version.

The level ends if we bring 3 masks to the hospital.
The hospital needed 2 colliders. One trigger collider
to check for the event that the player enters, but that
does not hinder the player from entering.
And a second collider that only collides with NPCs and keeps
them from entering the hospital.

Currently NPCs may still spawn inside the hospital and cant escape.

holke committed:
Arghhhh, this stupid unity editor did not save so
half of the stuff got lost during the last commit.
This is annoying -.-

holke committed:
Merge branch 'johannes/#103-nonspawnablespace' into johannes/levelmasks

holke committed:
Update. Masktext finds player correctly.
Set random grid

holke committed:
Added transparent UI for end conditions

holke committed:
Start text for mask level

holke committed:
Control populations size via public variable.
Change day length of infection control (was too short).
Change num masks to 8

holke committed:
Spawnable space only applies to active colliders

holke committed:
Fix bug in nonspawnable space. Infinte loop in player spawn

holke committed:
Use nonspawnable space for masks as well and extend nonspawnable space

holke committed:
nonspawnable update

holke committed:
Unity didnt safe, so guess this belongs to the prev commit

holke committed:
Add bool to levelsettings whether to show infection status or not

This makes it more feasible for levels different then gethome
to have their infection status shown or not

holke committed:
Start adding player house to map

Commit:
Added player house prefab to levelmasks

The player house
 * optional, not implemented yet: notifies the end level controller when the player is home
 * sets the player in the house depending on their condition
 * sets the player inside when space-key is pressed
 * sets the player outside when space-key is pressed

Commit:
Forgot to push scene

Commit:
Stop coroutines

OnTriggerExit all coroutines are stopped. We only want the player
to get into the house when he is actually near it.

holke committed:
Merge pull request #114 from youvsvirus/levelmasks/playerhouse

Added player house prefab to levelmasks

holke committed:
Add house pictures

holke committed:
Merge remote-tracking branch 'origin/johannes/levelmasks' into johannes/levelmasks

holke committed:
Merge remote-tracking branch 'origin/master' into johannes/levelmasks

holke committed:
try to repair merge

maccxs commented:
I will have a look

maccxs commented:
Is the plan that also for these levels, the player has to press space to enter the house?

Commit:
Fixed levelgethome

Commit:
Levelgethome fixed, forgot scene

Commit:
Fixed levelsupermarket

Commit:
Fix levelcollectmasks

maccxs commented:
All player houses fixed, probably the non-moving sprites in levelsupermarket are due to missing pathfinding project? Can you check if this works in the master for you?

holke commented:
Thanks.

Whatever you prefer.
In my opinion it felt very Natural in the get Home and Supermarkt Level That the player just enters the house without me needing to do anything.

maccxs commented:
I have done it like that

Am Fr., 22. Mai 2020 um 20:05 Uhr schrieb holke <notifications@github.com>:

> Thanks.
>
> Whatever you prefer.
> In my opinion Infekt very Natural in the get Home and Supermarkt Level
> That t player just enters the house without me needing to do anything.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/youvsvirus/youvsvirus-unity/pull/117#issuecomment-632833431>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALXB3RZKHONSN5XCV3RZ743RS25FPANCNFSM4NH7OUVQ>
> .
>

maccxs commented:
Your level is not easy. I really like it. :-) Is ist wanted that the player can move behind the hospital? And the NPCs cannot? Somehow we have to tell the user that they can hide in the house. Really cool!!! 
:-) :-) :-)

holke commented:
Thanks for fixing this :-)
Will check whether it works tomorrow.

I am happy that you like the level.
And yes it is intended that the player can walk „inside“ the hospital and not the npcs.

To be done:
- inform player that he can enter/exit the house 
- move the mask stat to the top right 
- ???

maccxs commented:
I like the idea that the player can walk into the hospital but during the game it feels like an error since you see the player vanish behind the hospital. Also, the player can hide behind the hospital before NPCs but not really because we do not see his position.  Maybe allow him to enter the hospital as well as his own house by pressing space? Otherwise there is nothing to be done. I will have a close look at it when you have finished?

holke commented:
Thank you for looking into this and fixing it.
The house now behaves as intended.

There are still 4-6 smileys in the supermarket level that are not moving, though.
I guess these are the bridge crossing smileys.
The pathfinding project folder is in the assets.
I also get debug messages like "Path completed. Computation time ... "

holke committed:
Gameplay improvements in mask level

- Player now collides with hostpital as well
- Counter is now in top right corner
- Slightly changed player house position

holke commented:
> 
> 
> I like the idea that the player can walk into the hospital but during the game it feels like an error since you see the player vanish behind the hospital. Also, the player can hide behind the hospital before NPCs but not really because we do not see his position. Maybe allow him to enter the hospital as well as his own house by pressing space? Otherwise there is nothing to be done. I will have a close look at it when you have finished?

I now changed it such that the player also collides with the hospital.
When all masks are collected and the player collides with the hospital the level finishes.

holke committed:
The player starts in the house now

holke committed:
Fixed a bug where we could win and loose at the same time.

The end level controller check for win/loose should now be
mutually exclusive.

Before it could happen that you had all masks, thus triggering the
succ canvas, but then got infected in the background, triggering
the fail canvas.
This should now not be possible anymore.
Once the level is won, it is won.

holke committed:
Edit space bar text

holke committed:
supermarket now continues to mask level

holke committed:
Reset number of needed mask to 8 (was 1 for debugging)

holke commented:
I would consider the mask level to be done now :)
Please check and give me some feedback.

What still needs to be fixed before merge are the non-moving NPCs in the supermarket level.

Another point:
It is cool that we can enter the house again to be safe for a while.
However, since the house is in the corner, we actually never are close to the house to make use of this feature.
Would it be worthwhile to change the map such that the house is more in the middle?
But, would that not involve having to deal with the rhine and the bridges?
So, maybe this is too much a change for now...
What is you opinion?

holke commented:
Actually, the NPCs in the supermarket are not moving in the master branch as well.
So this is propably not related to my merge efforts.

holke commented:
Not quite done yet.
I just noticed:

- The disco level is not called correcly at the end.
- It would make much more sense if the player needs to get back home after delivering the masks.

holke commented:
Also maybe remove the mask counter from the start screen?

holke committed:
Implement a script that can pause and unpause the game.

This can be useful for either
 - Binding keyboard buttons to pause/unpause
 - Use this to pause the game while displying information
   to the user

holke committed:
Merge branch 'johannes/PauseScript' into johannes/levelmasks

holke committed:
Prepare pausing the mask level when all masks are in hospital

holke committed:
Made all functions static

holke committed:
Merge branch 'johannes/PauseScript' into johannes/levelmasks

holke committed:
The player must return home after delivering the masks.

Added a new canvas that pauses the game and informs the
player to get home.

The level ends if the player returns home safely after delivering
the masks.

holke committed:
Update player house

the player house notifies the endlevelcontroller when
the player left.

Also tried to fix a bug where the player can be inside and outside
at the same time.
I made it such that the coroutines for entering and exiting should not be executed simultaneously,
however, the bug is still present.

holke committed:
The press space to continue script for the pause canvas

maccxs commented:
Wanted to continue my review, but player won't leave the house if I press space.

maccxs commented:
If i leave the house and go a bit aways I cannot get inside again
If i leave and do not move away I can get inside

maccxs commented:
Will you tell me when you want more feedback on this level, please?

holke commented:
Up to the sometimes buggy house and the continue button still leading to the wrong level, this is now really finished (final2_final.exe ;) ) and would appreciate more feedback.

Also, could you elaborate a bit more on the house behavior?
How far away do you go until it stops working? And what did you expect to happen instead?

maccxs commented:
If I press space to leave and then immediately press space again it works. If I move out of trigger zone and back in it does not.

maccxs commented:
I I go to the hospital without any masks, the canvas says "The hospital has enough masks now"

maccxs commented:
Can we maybe introduce a test mode for this level where you only have to collect one or two masks? It is quite hard to get all of them to test all functionalities.

maccxs commented:
Again and again :-)

maccxs commented:
Btw, I still love how hard it is and that I am really invested in getting all the masks, and in the end I am still getting infected. It's really annoying and addicting at the same time.

maccxs commented on YouVsVirus/Assets/Scenes/LevelCollectMasks/CreatePopLevelcollectmasks.cs:
```diff
@@ -0,0 +1,151 @@
+                // player is no longer shown in the house
+                PlayerHouse.GetComponent<PlayerHouse>().UnshowPlayer();
+                // can start to play
+                Player.gameObject.SetActive(true);
```
Did you not tell me that it is better to use .enabled ?

holke replied:
That was for the sprite renderer, where the alternative was setting it to a different layer.

I guess SetActive is the proper way to go here, since it activates the whole game object.
This way, the player just does not exist when he is inside the house.
Which is good, since otherwise it could still move around, even though invisible.

maccxs commented on YouVsVirus/Assets/Scenes/Levelgethome/CreatePopLevelgethome.cs:
```diff
@@ -44,7 +46,7 @@ public class CreatePopLevelgethome : MonoBehaviour
         /// Number of instantiated NPCs.
         /// </summary>
-        private int npcNumber = 42;
+        public int npcNumber = 42;
```
Why does this have to be public?

holke replied:
This way, we can easily control the number of NPCs from the unity scene view.
Which makes the whole script reusable for other levels.
Otherwise, we either have to copy the script, just to change the number. Or have to do
some "if name == 'MyLevel' " stuff which is error prone.

maccxs commented on YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs:
```diff
@@ -26,16 +46,19 @@ private void Start()
         endlevel = LevelSettings.GetActiveEndLevelController();
         // activate end level controller
-
+        endlevel = LevelSettings.GetActiveEndLevelController().GetComponent<EndLevelControllerBase>();
```
If we have this, the previous line is obsolete, is it not?

holke replied:
mhmmm, strange. I will investigate.

maccxs replied:
Maybe I did that ;-)

holke replied:
xD

holke commented:
> I I go to the hospital without any masks, the canvas says "The hospital has enough masks now"

Argh. I set the needed masks to 0 for debugging, but did not want to push this.

> Can we maybe introduce a test mode for this level where you only have to collect one or two masks? It is quite hard to get all of them to test all functionalities.

Yeah, thats why i set it to 0 for myself.
You can just change the number in the endlevelcontroller script.

holke commented on YouVsVirus/Assets/Scripts/Components/FaceMask.cs:
```diff
@@ -0,0 +1,61 @@
+        
+        void OnTriggerEnter2D (Collider2D other)
+        {
+            Debug.Log ("Collision detected.");
```
```suggestion
            Debug.Log ("Blub detected.");
```

maccxs commented:
If I play levelsupermarket, the game sometimes ends early and shows me the statistics end screen... ?

holke commented:
> If I play levelsupermarket, the game sometimes ends early and shows me the statistics end screen... ?

The one from sandbox mode?
mhmm, thats bad

maccxs commented:
> 
> 
> > If I play levelsupermarket, the game sometimes ends early and shows me the statistics end screen... ?
> 
> The one from sandbox mode?
> mhmm, thats bad

Yes, the one from sandbox

maccxs commented:
Maybe linked to pressing space? But not sure about that and not always

maccxs commented:
> 
> 
> Maybe linked to pressing space? But not sure about that and not always

No, not linked to space

holke commented:
I guess ill spend my weekend on a "finalfinal_now_finallydone.exe".

maccxs commented:
Mostly these seemingly deeply disturbing errors in the game are fixed really quickly in unity

holke committed:
Restore number of masks needed

holke committed:
Improve house. we can now enter and exit as we wish.
However, when we go away we cannot enter again :(

holke committed:
remove duplicate line

holke commented:
The house works better now.
However, the trigger enter seems broken.
Once we walk away, we cannot enter it :(

holke committed:
Update scenes with new house script

Commit:
Discussing coroutines

Commit:
Misleading comment

holke committed:
A new version of the house.

Coroutines are only startet whitin update
and only if no other coroutines run.

We have an additional trigger that is responsible
for telling the house whether the player is close or not.

holke committed:
Necessary changes to incorporate the new house

holke committed:
Unity did not save...again!...and again and again!.....all work and no save makes joe a dull boy, all work and no save makes joe a dull boy, all work and no save makes joe a dull boy, all work....

holke committed:
Finish the player house script

Added new var for when the player starts inisde.
A canvas can be given that is deactivated when the player
starts inside and leaves for the first time.

holke committed:
Use the new house in all levels

holke committed:
only debug log

holke commented:
Still need to activate the nonspawnable space for the mask.
Otherwise this is finished now.

holke committed:
Edit mask prefab

Has now the correct script and nonspawnable space attached.

holke committed:
build settings

holke committed:
remove part of code used for debugging

holke committed:
reset number of NPCs to correct value

holke commented:
I declare this to be finished.

maccxs commented:
If I am out of the house I cannot get back in

maccxs commented:
Why is Press Space called PressSpace(Missing Prefab)

maccxs commented:
Nooo, Cant' get out of the house now

maccxs commented:
Sorry, you already know I love your level :-)

holke commented:
> 
> 
> If I am out of the house I cannot get back in

Yes, that is supposed to happen.

holke commented:
> 
> 
> Nooo, Cant' get out of the house now

Are you inside before the level finished???
That´s not supposed to happen...
You should only be able to get inside when you have collected everything and the level should then finish...

holke committed:
Add pressspace prefab

maccxs commented:
I thought the player was supposed to be able to hide in the house?

holke commented:
As i said above, i deactivated this option. I think it confused the player more than it helped in this level.
And the player can still "hide" by just standing in the door.

holke commented:
Sorry, didnt say it above but in the house discussion thread.

maccxs commented on YouVsVirus/Assets/Scenes/LevelCollectMasks/CreatePopLevelcollectmasks.cs:
```diff
@@ -0,0 +1,158 @@
+
+        void Update()
+        {
+           // if (Input.GetKeyDown(KeyCode.Space))
```
Remove if no longer needed

maccxs commented on YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs:
```diff
@@ -6,61 +6,117 @@
+    // and set the player inside.
+    public void NotifyPlayerInside (bool IsInsideFlag)
+    {
+        // TODO: Set the player actually inside
```
What do you mean? Take the player himself?

holke replied:
No, i ment that this routine should also be able to display the sprite of the mini player in the window.
Which it does now, i just forgot to delete that comment.

maccxs commented on YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs:
```diff
@@ -71,23 +127,131 @@ public void UnshowPlayer()
+    /// Set the internal player object.
+    /// This should be called from the object that
+    /// instantiates the player. 
+    /// Reaon: Sometimes the player object
```
```suggestion
    /// Reason: Sometimes the player object
```

maccxs commented on YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs:
```diff
@@ -71,23 +127,131 @@ public void UnshowPlayer()
+    /// This should be called from the object that
+    /// instantiates the player. 
+    /// Reaon: Sometimes the player object
+    /// is not active but the house needs to avtivate it.
```
```suggestion
    /// is not active but the house needs to activate it.
```

maccxs commented on YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs:
```diff
@@ -71,23 +127,131 @@ public void UnshowPlayer()
+
+    void Update ()
+    {
+        // TODO: Only do this every 0.1 secs or so to increase performance
```
Remove comment?

holke replied:
```suggestion
```

maccxs commented on YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs:
```diff
@@ -71,23 +127,131 @@ public void UnshowPlayer()
+    void Update ()
+    {
+        // TODO: Only do this every 0.1 secs or so to increase performance
+        if (Time.time - lastTime >= timeBetweenUpdates)
```
Time.frameCount will give you the total number of frames since you started the game. so something like how many times the Update method was called since the start. Would it be better to use this instead of a time difference? Since frameCount is counted by unity anyway we would only need something like Time.frameCount%whatsoever. instead of save extra time variables?

holke replied:
Thanks for pointing this out, i didnt know that.
In some situations i would still prefer working with time, since it is independent of the frame rate...
Nevertheless, we can remove this whole if since it is only used for debugging output that is no longer needed.

maccxs commented on YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs:
```diff
@@ -71,23 +127,131 @@ public void UnshowPlayer()
+        }
+        if (player == null) {
+            // Do nothing is we do not have a reference of the player.
+            // This reference can be set withe the setPlayer function,
```
```suggestion
            // This reference can be set with the setPlayer function,
```

maccxs commented on YouVsVirus/Assets/Scenes/levelsupermarket/EndLevelControllerLevelsupermarket.cs:
```diff
@@ -26,4 +26,17 @@ protected override void CummulativeSpriteUpdate()
+        return player.GetComponent<Player>().hasToiletpaper;
+    }
+
+    // TODO: Why does this endlevelcontroller not know about the toilet paper? Having the toilet paper is a crucial condition for ending the game successfully.
```
In my version the player house was responsible for allowing the player to enter or not to enter depending on whether he has toilet paper or not

maccxs commented on YouVsVirus/Assets/Scripts/Components/FaceMask.cs:
```diff
@@ -0,0 +1,64 @@
+namespace Components
+{
+    /// The mask class does nothing, except being
+    /// a class that we can associate the mask sprit to.
```
```suggestion
    /// a class that we can associate the mask sprite to.
```

holke committed:
Typo

Co-authored-by: maccxs <49159623+maccxs@users.noreply.github.com>

holke committed:
typo

Co-authored-by: maccxs <49159623+maccxs@users.noreply.github.com>

holke committed:
Remove a comment

holke committed:
typo

Co-authored-by: maccxs <49159623+maccxs@users.noreply.github.com>

holke committed:
typo

Co-authored-by: maccxs <49159623+maccxs@users.noreply.github.com>

holke commented on YouVsVirus/Assets/Scenes/LevelCollectMasks/CreatePopLevelcollectmasks.cs:
```diff
@@ -0,0 +1,158 @@
+            NPCs[Mathf.RoundToInt(npcNumber/2f)].SetInitialCondition(NPC.EXPOSED);
+        }
+
+        void Update()
+        {
+           // if (Input.GetKeyDown(KeyCode.Space))
+           // {
+            //    // player is no longer shown in the house
+            //    playerHouse.GetComponent<playerHouse>().UnshowPlayer();
+            //    // can start to play
+           //    Player.gameObject.SetActive(true);
+           //     PressSpace.gameObject.SetActive(false);
+           // }
+        }
```
```suggestion
```

holke committed:
Remove House Update

holke commented on YouVsVirus/Assets/Scenes/Levelgethome/PlayerHouse.cs:
```diff
@@ -6,61 +6,117 @@
+    // and set the player inside.
+    public void NotifyPlayerInside (bool IsInsideFlag)
+    {
+        // TODO: Set the player actually inside
```
```suggestion
```

holke committed:
Delete comment

maccxs commented on YouVsVirus/Assets/Scripts/Components/Player.cs:
```diff
@@ -88,5 +93,24 @@ public override void SetSpriteImages()
 			DeadSprite = Resources.Load<Sprite>("SmileyPictures/player_dead");
 		}
 
+		void OnTriggerEnter (Collider other)
```
Still needed?

holke replied:
Nope

maccxs commented on YouVsVirus/Assets/Scripts/LevelCollectMasks/ButtonHandleLevelMasks.cs:
```diff
@@ -0,0 +1,29 @@
+        SceneManager.LoadScene("StartScreenLeveldisco_1");
+    }
+
+    public void UnshowCanvasAllCollectedGetHomeAndContinue()
```
maybe comment this function, maybe not

holke replied:
done

maccxs commented on YouVsVirus/Assets/Scripts/LevelCollectMasks/MaskCollect.cs:
```diff
@@ -0,0 +1,12 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class MaskCollect : MonoBehaviour
```
Artifact? Otherwise documentation needed

maccxs commented on YouVsVirus/Assets/Scripts/LevelCollectMasks/MaskContactTrigger.cs:
```diff
@@ -0,0 +1,14 @@
+
+namespace Components
+{
+    public class MaskContactTrigger : MonoBehaviour
```
Artifact? --> Delete

maccxs commented on YouVsVirus/Assets/Scripts/LevelCollectMasks/MaskControl.cs:
```diff
@@ -0,0 +1,74 @@
+        /// </summary>
+        private Vector2 screenBounds;
+
+        /// <summary>
+        /// The Sprites corresponding to the images for the different conditions
+        /// Images have to be set in the derived classes
+        /// </summary>     
+      //  protected Sprite MaskSprite=Resources.Load<Sprite>("Masks/mask");
```
```suggestion

```

maccxs commented on YouVsVirus/Assets/Scripts/LevelCollectMasks/MaskControl.cs:
```diff
@@ -0,0 +1,74 @@
+            float bufferZoneX = screenBounds[0]/10;
+            float bufferZoneY = screenBounds[1]/10;
+
+            // Create random coordinates and check if we can spawn a masks there.
```
```suggestion
            // Create random coordinates and check if we can spawn a mask there.
```

maccxs commented on YouVsVirus/Assets/Scripts/LevelCollectMasks/MaskControl.cs:
```diff
@@ -0,0 +1,74 @@
+            float bufferZoneY = screenBounds[1]/10;
+
+            // Create random coordinates and check if we can spawn a masks there.
+            // If not, we recreate.
```
```suggestion
            // If not, we try again.
```

maccxs commented on YouVsVirus/Assets/Scripts/LevelCollectMasks/hospital.cs:
```diff
@@ -0,0 +1,36 @@
+
+namespace Components
+{
+    public class hospital : MonoBehaviour
```
I think all our classes have capital letters at the beginning.

maccxs commented on YouVsVirus/Assets/Scripts/LevelCollectMasks/hospital.cs:
```diff
@@ -0,0 +1,36 @@
+                // Get number of masks from player
+                int numberOfMasks = other.GetComponent<Player>().getNumMasks();
+                // Get end level controller and notify him about the number of masks
+                EndLevelControllerBase elc = LevelSettings.GetActiveEndLevelController();
+                elc.NotifyInt (numberOfMasks);
```
Why have a variable elc here and not do it in one step?

holke replied:
Because:
1. It does not affect performance (the compiler optimizes it out)
2. It is in my opinion more readable compared to LevelSettings.GetActiveEndLevelController().NotifyInt (numberOfMasks)

maccxs commented on YouVsVirus/Assets/Scripts/LevelCollectMasks/hospitalCollider.cs:
```diff
@@ -0,0 +1,28 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class hospitalCollider : MonoBehaviour
```
Should be Capitalized

maccxs replied:
We do not really need this script, do we?

holke replied:
nope

maccxs commented on YouVsVirus/Assets/Scripts/UI/Levelcollectmasks/masktext.cs:
```diff
@@ -0,0 +1,72 @@
+namespace Components
+{
+
+public class masktext : MonoBehaviour
```
Should begin with a capital letter?

maccxs replied:
Wouldn't it be much easier if the FaceMask.cs would get the mask text as game object and could update it as soon as the player has the mask?

holke replied:
The mask itself does not know how many masks have been collected. And it may also change towards the other direction.
Example: The player collects 8 masks -> the counter is 8.
The player gets to the hospital and "gives it masks to the hospital" -> the player now has 0 masks -> the counter is 0.
However, the mask object itself cannot know that the player was at the hospital or not.
So the text is linked to the player´s inventory.

maccxs commented on YouVsVirus/Assets/Scenes/leveldisco/StartScreenLeveldisco_2.unity.meta:
```diff
@@ -1,5 +1,5 @@
 fileFormatVersion: 2
```
Is this scene or meta file still in use?

holke replied:
The mask itself does not know how many masks have been collected. And it may also change towards the other direction.
Example: The player collects 8 masks -> the counter is 8.
The player gets to the hospital and "gives it masks to the hospital" -> the player now has 0 masks -> the counter is 0.
However, the mask object itself cannot know that the player was at the hospital or not.
So the text is linked to the player´s inventory.

holke replied:
Yes. Its the Disco screen that tells the player "Jeff is your best friend..."

maccxs commented:
I think most scripts are now in the same folder as the scenes, you have separated it. What do you prefer?

maccxs commented:
I reall like the start screen ;-)

maccxs commented:
Should the text at the CanvasSucc get a comma? Congratulations, you made it! Or Congratulations! You made it!

maccxs commented:
Maybe rename both folders LevelCollectMasks to levelcollectmasks, I used this since you introduced it when renaming the scenes to StartscreenLevelcollectmasks, etc. ?

maccxs commented:
Oh no, the player is missing in the sandbox level now!

maccxs commented:
Again, to sum up: Awesome level and lots of work that went into it! I am amazed at your skills of thinking about code reusability and future strategies even in this new and unknown setting :-) Very good job. If we could have the player in the sandbox mode again and if the other minor comments are answered I will be very happy to merge this.

holke committed:
reactivate player in sandbox

holke commented:
> Oh no, the player is missing in the sandbox level now!

I just reactivated the player. Should work now.

holke committed:
Rename the levelcollectmaskfolder to lower case

holke committed:
remove old mask scripts that are not used anymore

holke commented:
> Wouldn't it be much easier if the FaceMask.cs would get the mask text as game object and could update it as soon as the player has the mask?

The mask itself does not know how many masks have been collected. And it may also change towards the other direction.
Example: The player collects 8 masks -> the counter is 8.
The player gets to the hospital and "gives it masks to the hospital" -> the player now has 0 masks -> the counter is 0.
However, the mask object itself cannot know that the player was at the hospital or not.
So the text is linked to the player´s inventory.

holke committed:
remove unused trigger from player

holke committed:
comment UnshowCanvasAllCollectedGetHomeAndContinue()

holke committed:
hospital -> Hospital

holke committed:
Remove old unneede script

holke committed:
rename hospital->Hospital

holke commented:
> I think most scripts are now in the same folder as the scenes, you have separated it. What do you prefer?

mhmm, i am a bit confused here.
We have a folder Scripts, which we should use for scripts.
But there are sometimes very level specific scripts und sometimes scripts that were introduced in a level but during develop get the status of scripts that are needed everywhere.

I would suggest that we properly discuss this again after the release and clean up our folder structure.

holke commented:
> 
> 
> Maybe rename both folders LevelCollectMasks to levelcollectmasks, I used this since you introduced it when renaming the scenes to StartscreenLevelcollectmasks, etc. ?

done

holke commented:
I think i am done with your comments.

Sorry if you get some deleted messages and general confusion stuff.
My github browser interface messed up and i sometimes wrote large comments that were then pasted in the wrong place, so i had to do a lot of reediting.

holke commented:
> 
> 
> Should the text at the CanvasSucc get a comma? Congratulations, you made it! Or Congratulations! You made it!

Is now "Congratulations, you made it!"
